test/integration/clustercidr/ipam_test.go

```diff
@@ -0,0 +1,246 @@
+
+	ipamController := booststrapMultiCIDRRangeAllocator(t, clientSet, sharedInformer)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go ipamController.Run(ctx.Done())
+	sharedInformer.Start(ctx.Done())
+
+	tests := []struct {
+		name             string
+		clusterCIDR      *networkingv1alpha1.ClusterCIDR
+		node             *v1.Node
+		expectedPodCIDRs []string
+	}{
+		{
+			name:             "Default dualstack Pod CIDRs assigned to a node, node labels matching no ClusterCIDR nodeSelectors",
+			clusterCIDR:      nil,
+			node:             makeNode("default-node", map[string]string{"label": "unmatched"}),
+			expectedPodCIDRs: []string{"10.96.0.0/24", "fd00:10:96::/120"},
+		},
+		{
+			name:             "Dualstack Pod CIDRs assigned to a node from a CC created during bootstrap",
+			clusterCIDR:      nil,
+			node:             makeNode("bootstrap-node", map[string]string{"bootstrap": "true"}),
+			expectedPodCIDRs: []string{"10.2.1.0/24", "fd00:20:96::100/120"},
+		},
+		{
+			name:             "Single stack IPv4 Pod CIDR assigned to a node",
+			clusterCIDR:      makeClusterCIDR("ipv4-cc", "10.0.0.0/16", "", nodeSelector(map[string][]string{"ipv4": {"true"}, "singlestack": {"true"}})),
+			node:             makeNode("ipv4-node", map[string]string{"ipv4": "true", "singlestack": "true"}),
+			expectedPodCIDRs: []string{"10.0.0.0/24"},
+		},
+		{
+			name:             "Single stack IPv6 Pod CIDR assigned to a node",
+			clusterCIDR:      makeClusterCIDR("ipv6-cc", "", "fd00:20:100::/112", nodeSelector(map[string][]string{"ipv6": {"true"}})),
+			node:             makeNode("ipv6-node", map[string]string{"ipv6": "true"}),
+			expectedPodCIDRs: []string{"fd00:20:100::/120"},
+		},
+		{
+			name:             "DualStack Pod CIDRs assigned to a node",
+			clusterCIDR:      makeClusterCIDR("dualstack-cc", "192.168.0.0/16", "fd00:30:100::/112", nodeSelector(map[string][]string{"ipv4": {"true"}, "ipv6": {"true"}})),
+			node:             makeNode("dualstack-node", map[string]string{"ipv4": "true", "ipv6": "true"}),
+			expectedPodCIDRs: []string{"192.168.0.0/24", "fd00:30:100::/120"},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if test.clusterCIDR != nil {
+				// Create the test ClusterCIDR
+				if _, err := clientSet.NetworkingV1alpha1().ClusterCIDRs().Create(context.TODO(), test.clusterCIDR, metav1.CreateOptions{}); err != nil {
+					t.Fatal(err)
+				}
+
+				// Wait for the ClusterCIDR to be created
+				if err := wait.PollImmediate(time.Second, 5*time.Second, func() (bool, error) {
+					cc, err := clientSet.NetworkingV1alpha1().ClusterCIDRs().Get(context.TODO(), test.clusterCIDR.Name, metav1.GetOptions{})
+					if err != nil {
+						return false, err
+					}
+					return cc != nil, nil
+				}); err != nil {
+					t.Fatalf("failed while waiting for ClusterCIDR %q to be created: %v", test.clusterCIDR.Name, err)
+				}
```

Comment:
this may be racy here, since the creation of the ClusterCIDR() doesn't guarantee the controller has processed it when we create the node, for now I'm fine if you add something like this before creating the new node

```
	// Sleep for one second to make sure the controller process the new created ClusterCIDR
	time.Sleep(1 * time.Second)
```

Reply:
Done, makes sense